src/gs-app.c

```diff
@@ -2093,6 +2100,12 @@ gs_app_set_metadata (GsApp *app, const gchar *key, const gchar *value)
 	g_hash_table_insert (app->metadata,
 			     g_strdup (key),
 			     g_string_free (str, FALSE));
+
+	g_signal_emit (app,
+	               signals[SIGNAL_METADATA_ADDED],
```

Comment:
I'd call this `metadata-changed` and also call it when a key is removed (since replacing it does not seem a valid operation).